### PR TITLE
More ROP classification patches

### PR DESCRIPTION
**Pull request by Darredevil**
Patches for #5397

### CODE REVIEW

File: libr/core/cmd_search_rop.c [hunk 1]
```diff
@@ -181,9 +181,28 @@ static void fillRegisterValues (RCore *core) {
...
 
+// split esil string in flags part and main instruction
+// hacky, only tested for x86, TODO: portable version
+static void esil_split_flg (char *esil_str, char **esil_main, char **esil_flg) {
+	char *split = strstr (esil_str, "f,=");
+	int hits;
+
+	hits = 0;
+	if (split) {
+		while (hits != 2) {
```
Comment (crowell) [inline libr/core/cmd_search_rop.c]: what is this magic `2`? Can you use a constant that describes it?

File: libr/core/cmd_search_rop.c [hunk 2]
```diff
@@ -181,9 +181,28 @@ static void fillRegisterValues (RCore *core) {
...
+			--split;
+			if (*split == ',') {
+				hits++;
+			}
+		}
+		*esil_flg = strdup (++split);
```
Comment (crowell) [inline libr/core/cmd_search_rop.c]: put a comment for the function about the ownership of these strings.

File: libr/core/cmd_search_rop.c [hunk 3]
```diff
@@ -181,9 +181,28 @@ static void fillRegisterValues (RCore *core) {
 	r_reg_arena_push (core->dbg->reg);
 }
 
+// split esil string in flags part and main instruction
+// hacky, only tested for x86, TODO: portable version
+static void esil_split_flg (char *esil_str, char **esil_main, char **esil_flg) {
+	char *split = strstr (esil_str, "f,=");
+	int hits;
```
Comment (radare) [inline libr/core/cmd_search_rop.c]: assign and define in the same line

**Comment by Darredevil** [comment]
@crowell @radare comments fixed, ready for merge

File: libr/core/cmd_search_rop.c [hunk 4]
```diff
@@ -7,6 +7,8 @@
 #include "r_list.h"
 #include "r_types_base.h"
 
+#define COMA_HITS 2
```
Comment (crowell) [inline libr/core/cmd_search_rop.c]: not a define, do a `const int kCommaHits = 2`

also, what does it mean when you hit 2 commas?

File: libr/core/cmd_search_rop.c (hunk 2 above)
Comment (Darredevil) [inline libr/core/cmd_search_rop.c]: this is fixed, not sure why the diff isn't seeing it

File: libr/core/cmd_search_rop.c (hunk 4 above)
Comment (Darredevil) [inline libr/core/cmd_search_rop.c]: fixed again :)
what does it mean: initially you search the string for the first flag assignment, this marks the beginning of the flag only part, but you need to go back 2 commas to include the value for the first flag as well

**Pull request merged**